server/src/main/java/org/elasticsearch/index/mapper/XContentDataHelper.java

```diff
@@ -74,6 +84,58 @@ static void decodeAndWrite(XContentBuilder b, BytesRef r) throws IOException {
         }
     }
 
+    /**
+     * Stores the current parser structure (subtree) to an {@link XContentBuilder} and returns it, along with a
+     * {@link DocumentParserContext} wrapping it that can be used to reparse the subtree.
+     * The parser of the original context is also advanced to the end of the current structure (subtree) as a side effect.
+     */
+    static Tuple<DocumentParserContext, XContentBuilder> cloneSubContext(DocumentParserContext context) throws IOException {
```

Comment:
Maybe these methods instead of returning a tuple should return a record? I think `getParserContext()` or `getBuilder()` reads easier than `v1()` and `v2()`?

Reply:
Since we just need to wrap a couple of objects and there are no methods, I'd think a tuple is a good match here.

Reply:
Then this is just me :)

Reply:
I did change the call sites to skip using `var`, it's should be more readable now.